src/ui/components/Events/Event.tsx

```diff
@@ -31,15 +162,142 @@ export const getEventLabel = (event: ReplayEvent) => {
   return label;
 };
 
+const FIND_MOUSE_EVENT_TARGET_AND_REACT_CLICK_HANDLER = `
+  // Outer body runs in scope of the "current" click handler.
+  // Grab the click handler's arguments.
+  const args = [...arguments];
+  
+  (function findEventTargetAndClickHandler() {
+    // One of the args should be a mouse event
+    const foundMouseEvent  = args.find(a => typeof a === "object" && a instanceof MouseEvent);
+
+    if (foundMouseEvent) {
+      // We _could_ probably just return the onClick function or undefined here
+      const res = {
+        target: event.target,
+      }
+
+      // Search the event target node and all of its ancestors
+      // for React internal props data, and specifically look
+      // for the nearest node with an onClick prop if any.
+      let currentNode = event.target;
+      while (currentNode) {
+        const keys = Object.keys(currentNode)
+        const reactPropsKey = keys.find(key => {
+          return (
+            key.startsWith("${REACT_16_EVENT_LISTENER_PROP_KEY}") || 
+            key.startsWith("${REACT_17_18_EVENT_LISTENER_PROP_KEY}")
+          )
+        });
+
+        if (reactPropsKey) {
+          const props = currentNode[reactPropsKey];
+          if (props.onClick) {
```

Comment:
Do we care about `onClickCapture` ?

Reply:
Nope. Not trying to deal with that for now - it's extremely rarely used in my experience.